components/places/src/places.udl

```diff
@@ -1,31 +1,56 @@
-syntax = "proto2";
-
-// Note: this file name must be unique due to how the iOS megazord works :(
-
-package mozilla.appservices.places.protobuf;
-
-option java_package = "mozilla.appservices.places";
-option java_outer_classname = "MsgTypes";
-option swift_prefix = "MsgTypes_";
-option optimize_for = LITE_RUNTIME;
-
-message HistoryVisitInfo {
-    required string url = 1;
-    optional string title = 2;
-    required int64 timestamp = 3;
-    required int32 visit_type = 4;
-    required bool is_hidden = 5;
-}
-
-message HistoryVisitInfos {
-    repeated HistoryVisitInfo infos = 1;
-}
+//
+// UniFFI interface definition for the "places" component.
+//
+// The "places" component is still in the process of being converted
+// to UniFFI, so this file doesnt' define the entire interface.
+// For now it only contains the rich datatypes that we want to
+// transfer across the FFI boundary.
+
+
+namespace places {};
+
+enum VisitTransition {
+    /** This transition type means the user followed a link. */
+    "Link",
+    /** This transition type means that the user typed the page's URL in the
+     *  URL bar or selected it from UI (URL bar autocomplete results, etc).
+     */
+     "Typed",
+     "Bookmark",
+     "Embed",
+     "RedirectPermanent",
+     "RedirectTemporary",
+     "Download",
+     "FramedLink",
+     "Reload"
+};
+
+// Information about a specific history visit.
+//
+dictionary HistoryVisitInfo {
+    string url;
+    string? title;
+    i64 timestamp;
+    VisitTransition visit_type;
+    boolean is_hidden;
+};
+
+// A list of history visits, along with pagination info.
+// This is useful when loading lots of history visits via pagination.
+//
+dictionary HistoryVisitInfosWithBound {
+    sequence<HistoryVisitInfo> infos;
+    i64 bound;
+    i64 offset;
+};
+
+
+enum BookmarkType {
```

Comment:
I notice alot of the diff above is just the case changing of this enum - is there a particular reason you didn't leave the old names?

Reply:
Ugh, yeah, UniFFI automagically uppercases enum variant names in the Kotlin bindings, since this seems to be the prevailing style convention for Kotlin code.